pkg/agent/clean/node.go

```diff
@@ -0,0 +1,349 @@
+// +build !windows
+
+package clean
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/coreos/go-iptables/iptables"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/filters"
+	"github.com/docker/docker/api/types/network"
+	"github.com/docker/docker/client"
+	"github.com/rancher/rke/hosts"
+	"github.com/sirupsen/logrus"
+	"github.com/vishvananda/netlink"
+)
+
+const ContainerName = "cattle-node-cleanup"
+
+func Job() error {
+	logrus.Infof("Starting clean container job: %s", ContainerName)
+
+	c, err := client.NewClientWithOpts(client.FromEnv)
+	if err != nil {
+		return err
+	}
+	defer c.Close()
+
+	ctx := context.Background()
```

Comment:
I think a general good practice would be to create the context in main and pass it in via the function signature right?